### PR TITLE
feat(list-item): Add content slot for specialized content

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #3032

## Summary

feat(list-item): Add slot for specialized content #3032

- Adds a new `content` slot to the `list-item` for customized content.
- `content` slot takes precedence over `label` and `description` properties.

### CODE REVIEW

File: src/components/list-item/list-item.tsx [hunk 1]
```diff
@@ -33,6 +33,7 @@ import {
  * @slot - A slot for adding `calcite-list-item` and `calcite-list-item-group` elements.
  * @slot actions-start - A slot for adding actionable `calcite-action` elements before the content of the component.
  * @slot content-start - A slot for adding non-actionable elements before the label and description of the component.
+ * @slot content - A slot for adding non-actionable centered content in place of the label and description of the component.
```
Comment (benelan) [inline src/components/list-item/list-item.tsx:36]: Can you change the description to

> A slot for adding non-actionable, centered content in place of the \`label\` and \`description\` of the component.

**Review by benelan — CHANGES_REQUESTED** [review]
Something is funky with the focus when using the `content` slot. You can repro in the story after removing the `disabled` attributes, or put this at the end of the list demo:

```html
    <!-- custom content -->
    <div class="parent">
      <style>
        .list-chip {
          margin-inline-end: 0.5rem;
        }
      </style>

      <div class="child right-aligned-text">custom content</div>
      <div class="child">
        <calcite-list>
          <calcite-list-item>
            <div slot="content">
              <strong>Cras iaculis ultricies nulla.</strong>
              <div>Lorem ipsum dolor sit amet, consectetuer adipiscing elit.</div>
            </div>
          </calcite-list-item>
          <calcite-list-item>
            <div slot="content">
              <strong>Cras iaculis ultricies nulla.</strong>
              <div>Lorem ipsum dolor sit amet, consectetuer adipiscing elit.</div>
            </div>
          </calcite-list-item>
          <calcite-list-item>
            <div slot="content">
              <strong>Cras iaculis ultricies nulla.</strong>
              <div>Lorem ipsum dolor sit amet, consectetuer adipiscing elit.</div>
            </div>
          </calcite-list-item>
        </calcite-list>
      </div>
    </div>
```

For the most part, I can't get the focus ring to show up when clicking or tabbing. However if you click/tab for long enough the focus ring will eventually show up on one of the items. I also noticed some inconsistent focus ring behavior for the "start/end content slots" demo, so it might not be completely related to this PR. But the content slot focus behavior seems a lot worse

**Comment by driskull** [comment]
> Something is funky with the focus when using the content slot. You can repro in the story after removing the disabled attributes, or put this at the end of the list demo:

@benelan can you create an issue for this one? I do see the problem.

**Review by benelan — APPROVED** [review]
Yeah I confirmed it's reproducible on our `next` storybook deployment, so it isn't a result of the changes in this PR. 
https://esri.github.io/calcite-components/?path=/story/components-list--start-and-end-content-slots

If you rather address it in a different PR I'm okay with that; I'll create an issue. Other than that and the doc nitpick, LGTM

**Comment by driskull** [comment]
> If you rather address it in a different PR I'm okay with that; I'll create an issue. Other than that and the doc nitpick, LGTM

Yeah, it would be easier to track and report that way 👍

**Pull request merged**